### PR TITLE
largest component size by common factor added

### PR DESCRIPTION
**Pull request by gayatri517**

### CODE REVIEW

**Review by priyanshisharma — CHANGES_REQUESTED** [review]
* Can you please rename this file to `August_30_Largest_Component_Size_by_Common_Factor.cpp`.
* Please write a function for the following header, just like you would submit it on leetcode.
```
class Solution {
public:
    int largestComponentSize(vector<int>& A) {
        
    }
};
```

**Comment by gayatri517** [comment]
Hi @priyanshisharma i have made the changes kindly review them.

File: August/August_30_Largest_Component_Size_by_Common_Factor.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        dsu DSU(n+1);
+        for(auto x:A){
+            int temp=sqrt(x);
+            for(int k=2;k<=temp;k++){
+                if(x%k==0){
+                    DSU.merge(x,k);
+                    DSU.merge(x,x/k);
+                }
+            }
+        }3
```
Comment (priyanshisharma) [inline August/August_30_Largest_Component_Size_by_Common_Factor.cpp:39]:
```suggestion
        }
```

**Review by priyanshisharma — CHANGES_REQUESTED** [review]
Beats 80.88% after that tiny removal, please do the same, i'll be happy to merge this 🚀

**Comment by gayatri517** [comment]
@priyanshisharma i have made that change kindly review it.

**Review by priyanshisharma — APPROVED** [review]
Looks perfect ✨ Merging this 🚀

**Pull request merged**